packages/dx-vue-grid/src/plugins/table-group-row.test.js

```diff
@@ -0,0 +1,362 @@
+      render() {
+        return (
+          <PluginHost>
+            <PluginDepsToComponents deps={defaultDeps} />
+            <TableGroupRow
+              {...{ attrs: { ...defaultProps } }}
+              indentCellComponent={indentCellComponent}
+            />
+          </PluginHost>
+        );
+      },
+    });
+
+    expect(isGroupIndentTableCell)
+      .toBeCalledWith(
+        defaultDeps.template.tableCell.tableRow,
+        defaultDeps.template.tableCell.tableColumn,
+        defaultDeps.getter.grouping,
+      );
+    expect(tree.find(indentCellComponent).vm.$attrs)
+      .toMatchObject({
+        ...defaultDeps.template.tableCell,
+        row: defaultDeps.template.tableCell.tableRow.row,
+        column: defaultDeps.template.tableCell.tableColumn.column,
+      });
+  });
+
+  describe('cellComponent', () => {
+    it('should render group cell on select group column and group row intersection', () => {
+      isGroupTableRow.mockImplementation(() => true);
+      isGroupTableCell.mockImplementation(() => true);
+
+      const tree = mount({
+        render() {
+          return (
+            <PluginHost>
+              <PluginDepsToComponents deps={defaultDeps} />
+              <TableGroupRow
+                {...{ attrs: { ...defaultProps } }}
+              />
+            </PluginHost>
+          );
+        },
+      });
+
+      expect(isGroupTableCell)
+        .toBeCalledWith(
+          defaultDeps.template.tableCell.tableRow,
+          defaultDeps.template.tableCell.tableColumn,
+        );
+      expect(tree.find(defaultProps.cellComponent).vm.$attrs)
+        .toMatchObject({
+          ...defaultDeps.template.tableCell,
+          row: defaultDeps.template.tableCell.tableRow.row,
+          column: defaultDeps.template.tableCell.tableColumn.column,
+        });
+    });
+
+    it('should provide correct cell params', () => {
+      isGroupTableRow.mockImplementation(() => true);
+      isGroupTableCell.mockImplementation(() => true);
+
+      const deps = {
+        getter: {
+          expandedGroups: [],
+        },
+        template: {
+          tableCell: {
+            tableRow: { row: { compoundKey: '1' } },
+            tableColumn: { column: { name: 'a' } },
+          },
+        },
+      };
+      jest.spyOn(deps.getter.expandedGroups, 'indexOf').mockReturnValue(1);
+
+      const tree = mount({
+        render() {
+          return (
+            <PluginHost>
+              <PluginDepsToComponents deps={{ ...defaultDeps, ...deps }} />
+              <TableGroupRow
+                {...{ attrs: { ...defaultProps } }}
+              />
+            </PluginHost>
+          );
+        },
+      });
+      expect(tree.find(defaultProps.cellComponent).vm.$attrs)
+        .toMatchObject({
+          expanded: true,
+        });
+      expect(deps.getter.expandedGroups.indexOf)
+        .toBeCalledWith('1');
+
+      tree.find(defaultProps.cellComponent).vm.$listeners.toggle();
```

Comment:
Can we call trigger here?

Reply:
We [can't use trigger](https://github.com/vuejs/vue-test-utils/issues/145) here. But we can use `$emit`.